core/lib/snapshots_applier/src/tests.rs

```diff
@@ -0,0 +1,234 @@
+#[cfg(test)]
+mod utils {
+    use std::collections::HashMap;
+
+    use async_trait::async_trait;
+    use zksync_types::{
+        api::en::SyncBlock,
+        block::L1BatchHeader,
+        commitment::{L1BatchMetaParameters, L1BatchMetadata, L1BatchWithMetadata},
+        snapshots::{SnapshotHeader, SnapshotStorageLog},
+        AccountTreeId, L1BatchNumber, MiniblockNumber, StorageKey, StorageValue, H160, H256,
+    };
+    use zksync_web3_decl::jsonrpsee::core::ClientError as RpcError;
+
+    use crate::SnapshotsApplierMainNodeClient;
+
+    #[derive(Debug, Default)]
+    pub(crate) struct MockMainNodeClient {
+        pub(crate) fetch_l2_block_responses: HashMap<MiniblockNumber, SyncBlock>,
+        pub(crate) fetch_newest_snapshot_response: Option<SnapshotHeader>,
+    }
+
+    #[async_trait]
+    impl SnapshotsApplierMainNodeClient for MockMainNodeClient {
+        async fn fetch_l2_block(
+            &self,
+            number: MiniblockNumber,
+        ) -> Result<Option<SyncBlock>, RpcError> {
+            if let Some(response) = self.fetch_l2_block_responses.get(&number) {
+                Ok(Some((*response).clone()))
+            } else {
+                Ok(None)
+            }
```

Comment:
This can be shortened to `Ok(self.fetch_l2_block_responses.get(&number).cloned())`. Ditto for the below method – can be shortened to `Ok(self.fetch_newest_snapshot_response.clone())`.